lib/cc/cli/engines/engine_command.rb

```diff
@@ -24,6 +24,7 @@ def update_yaml
           File.open(filesystem.path_for(CODECLIMATE_YAML), "w") do |f|
```

Comment:
Not something to change now, but worth noting. I think there's an abstraction issue here. We go through `filesystem` for things like `path_for` or `stat`, but then we break the abstraction and call things like `File.open` or `File.write` directly? If `FileSystem` had a `#write` method that would be better, and `#path_for` could probably be made private as it'd only be used internally within (methods like) `FileSystem#write`.

Again, something to consider in a separate PR -- but it provides a little context for my `FileSystem#stat` suggestion.